src/python/pants/backend/docker/target_types.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from pants.engine.target import COMMON_TARGET_FIELDS, Dependencies, Sources, StringField, Target
+
+
+class DockerImageSources(Sources):
+    default = ("Dockerfile",)
+    expected_num_files = 1
+    help = "Name of the Dockerfile to use when building the Docker image."
+
+
+class DockerImageVersion(StringField):
+    alias = "version"
+    default = "latest"
+    help = "Image tag to apply to built images."
+
+
+class DockerDependencies(Dependencies):
+    pass
+
+
+class DockerContextRoot(StringField):
+    alias = "context_root"
+    default = "."
+    help = (
+        "Root directory for the Docker build context. Default is to use the directory of the "
```

Comment:
Formatting nit: probably good to use `\n\n` between these so that they're new paragraphs.